feat: Add clientId field to getClusterNodes response

leafaar opened the pull request:
#### Problem

The Agave RPC now returns a `clientId` field in the getClusterNodes response (see anza-xyz/agave#7817), but the JavaScript client doesn't include this field in its type definitions.

#### Summary of Changes

- Add `clientId: number | null` field to the `ClusterNode` type
- Update tests to expect the clientId field in responses
- Add JSDoc documentation for the new field

#### References

- Depends on: anza-xyz/agave#7817

leafaar committed:
feat: Add clientId field to getClusterNodes response

leafaar committed:
chore: Add changeset for clientId field

steveluscher reviewed (changes requested):
Make sure to update all of the tests (run `pnpm dev` in `packages/rpc-api/` as you work) and to add an exemption for `clientId` so that the client doesn't presume it's supposed to be a `BigInt`.

See #521 for a great example of how to do all of that!

leafaar committed:
test: Update tests and add BigInt exemption for clientId

leafaar commented:
Hey, I updated and fixed a lint, where the `ClusterNode` need to be in alphabetical order, so I moved `clientId` to the top. Also added the exception, so the client doesn't interpret as a BigInt. Sorry for the delay to fix that, I'm not too familiar with this part of the stack. Hope I did everything correct, if not, just let me know, and I'll be happy to fix it again :D

mcintyre94 commented:
Sorry @leafaar - I think we should have marked this do-not-close so the bot doesn't close it. Reopened and added that label, hopefully it'll be left alone now! 

We have a github action that closes stale issues/PRs automatically: https://github.com/anza-xyz/kit/blob/main/.github/workflows/manage-stale-threads.yml
